odoo/tools/config.py

```diff
@@ -792,11 +794,12 @@ def _check_path(cls, option, opt, value):
 
     @classmethod
     def _check_without_demo(cls, option, opt, value):
+        # invert the result because it is stored in "with_demo"
```

Comment:
For files, we call this function without `option`, so we could add a deprecation warning such as:
```
if option is not None:
  warnings.warn("Since 19.0, --without-demo option is deprecated, use --with-demo when needed")
```